### PR TITLE
add a value_object to represent search_controller/feed_content params

### PR DESCRIPTION
**Pull request by carlosdanielpohlod**
…ent params

## What type of PR is this? (check all applicable)

- [ x ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Add value object on search_controller#feed_content to decrease controller responsibilities and give more descriptive names for each conditional statement. Also, one of the conditional statements was removed due is unnecessary.

## Related Tickets & Documents

- Related Issue #
- Closes #

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, a note
on the devices and browsers this has been tested on, as well as any relevant
images for UI changes._

### UI accessibility concerns?

_If your PR includes UI changes, please replace this line with details on how
accessibility is impacted and tested. For more info, check out the
[Forem Accessibility Docs](https://developers.forem.com/frontend/accessibility)._

## Added/updated tests?

- [ ] Yes
- [ ] No, and this is why: _please replace this line with details on why tests
      have not been included_
- [ ] I need help with writing tests

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

**Comment by maestromac** [comment]
Hey @carlosdanielpohlod, how is it going with this PR? how can we help?

**Comment by carlosdanielpohlod** [comment]
hey @maestromac , I will reopen this pull request. Maybe you saw other pull requests by @gabrielcicero45  with similar changes. This happened because I was helping him to contribute to this project and we have a small confusion during the pull request, so we will close other pull requests and we will consider just this one, thanks for your comprehension

**Comment by mirie** [comment]
@carlosdanielpohlod Thank you for your contribution! It looks like we have a pending CLA for you. Do you mind signing it? Thanks!

**Comment by lboogie-ftf** [comment]
@carlosdanielpohlod This is a nice refactor! Thanks for this contribution. Upon signing the pending CLA, I'll return and add my insights/approval.

**Comment by carlosdanielpohlod** [comment]
@lboogie-ftf @mirie CLI signed!

File: app/value_objects/search_resources/feed_content.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+module SearchResources
```
Comment (maestromac) [inline app/value_objects/search_resources/feed_content.rb:1]: I see that you are introducing a new "value_object" pattern to our codebase. This is something that would need our eng team's discussion and buy-in. For this moment, can you please move this content to our `app/services/search` folder? thank you.

**Review by maestromac — CHANGES_REQUESTED** [review]

**Comment by mirie** [comment]
Hi @carlosdanielpohlod please let us know if you have any questions regarding the changes requested. Thanks again for moving this contribution forward!!

**Comment by gabrielcicero45** [comment]
@maestromac  I've made the changes you requested. Whenever you have a moment, please take a look.

**Comment by rt4914** [comment]
@maestromac Please have a look at previous comments and the latest changes by author. Thank you.

**Review by maestromac — CHANGES_REQUESTED** [review]
Thanks again for this PR. Upon closer look, I'm going to have to reject this PR at this state. 

There is value in `value_object` but I don't think refactor is an improvement. The class `Search::FeedContent` exist only to answer `is_homepage_search`, which in this case I would argue would've been better to remain in its current location. I think that it's the wrong focus on what should be refactored. My suggestion for bigger improvement would be to extract the entirety of `#feedcontent`. Follow the [Tell, don't ask](https://thoughtbot.com/blog/tell-dont-ask) principle to refactor out long `if` chains.

You are welcome to continue on this PR or start fresh in a new one.

**Comment by mirie** [comment]
Hello @carlosdanielpohlod Thank you again for your contribution! I'm going to close this PR for now per the last comment. You're welcome to pick this up again in the future~!

**Pull request closed without merging**